src/neural/xla/onnx2hlo.cc

```diff
@@ -313,45 +334,90 @@ class Onnx2HloConverter {
     auto* conv = builder_.Convolution(input, kernel, window, dn);
 
     if (!bias) return {conv};
-    auto* flow = builder_.Broadcast(bias, conv->shape(), {1});
+    auto* flow = builder_.Broadcast(bias, HloTensorType(conv->shape()), {1});
     return {builder_.Add(conv, flow)};
   }
 
   std::vector<HloFlow> OpRelu(const pblczero::NodeProto& node) {
-    CheckKnownAttributes(node, {});
+    CheckKnownAttributes(node, 1, {});
     auto* input = GetInput(node, 0);
     auto* zero = MakeScalar(0, input->shape().element_type());
-    zero = builder_.Broadcast(zero, input->shape(), {});
+    zero = builder_.Broadcast(zero, HloTensorType(input->shape()), {});
     return {builder_.Maximum(input, zero)};
   }
 
+  std::vector<HloFlow> OpSigmoid(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 1, {});
+    auto* input = GetInput(node, 0);
+    auto* one = MakeScalar(1, input->shape().element_type());
+    one = builder_.Broadcast(one, HloTensorType(input->shape()), {});
+    return {builder_.Divide(
+        one, builder_.Add(one, builder_.Exponential(builder_.Negate(input))))};
+  }
+
   std::vector<HloFlow> OpTanh(const pblczero::NodeProto& node) {
-    CheckKnownAttributes(node, {});
+    CheckKnownAttributes(node, 1, {});
     auto* input = GetInput(node, 0);
     return {builder_.Tanh(input)};
   }
 
   std::vector<HloFlow> OpAdd(const pblczero::NodeProto& node) {
-    CheckKnownAttributes(node, {});
+    CheckKnownAttributes(node, 2, {});
     auto* lhs = GetInput(node, 0);
     auto* rhs = GetInput(node, 1);
     std::tie(lhs, rhs) = EqualizeShape(lhs, rhs);
     return {builder_.Add(lhs, rhs)};
   }
 
-  std::vector<HloFlow> OpReshape(const pblczero::NodeProto& node) {
-    CheckKnownAttributes(node, {});
+  std::vector<HloFlow> OpMul(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 2, {});
+    auto* lhs = GetInput(node, 0);
+    auto* rhs = GetInput(node, 1);
+    std::tie(lhs, rhs) = EqualizeShape(lhs, rhs);
+    return {builder_.Multiply(lhs, rhs)};
+  }
+
+  std::vector<HloFlow> OpSplit(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 1, {"axis", "num_outputs"});
```

Comment:
`Split` is different before opset 13, there was a "split" attribute instead of input: https://github.com/onnx/onnx/blob/main/docs/Changelog.md#attributes-187
Also "num_outputs" was only added in opset 18.

Reply:
I've just added a check for the opcode version, to show that it's possible in principle to have different versions (what I'd do is to make `OpSplit()` call `OpSplit1()`, `OpSplit13()` etc depending on the version).

But for now the code just fails.